src/Cuboid.cpp

```diff
@@ -24,10 +24,10 @@ static bool DoIntervalsIntersect(int a_Min1, int a_Max1, int a_Min2, int a_Max2)
 ////////////////////////////////////////////////////////////////////////////////
 // cCuboid:
 
-cCuboid & cCuboid::operator =(cCuboid a_Other)
+cCuboid & cCuboid::operator =(const cCuboid & a_Other)
```

Comment:
Performance wise, I think (without having measured it) it's the same order of magnitude as passing Vectors as reference vs value.

Readability wise, I think it's much more obvious what's going on with the equal signs than with the swaps.